### ISSUE TITLE
TensorBoard graph visualization failure

### ISSUE THREAD

**Issue by kssion** [ultralytics/ultralytics#23265]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug
```
Ultralytics 8.4.3 🚀 Python-3.11.14 torch-2.9.1+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=1, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=dota8.yaml, degrees=0.0, deterministic=True, device=None, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=10, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=320, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolo26n-obb.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=train4, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, perspective=0.0, plots=True, pose=12.0, pretrained=True, profile=False, project=None, rect=False, resume=False, retina_masks=False, rle=1.0, save=True, save_conf=False, save_crop=False, save_dir=/yolo/runs/obb/train4, save_frames=False, save_json=False, save_period=-1, save_txt=False, scale=0.5, seed=0, shear=0.0, show=False, show_boxes=True, show_conf=True, show_labels=True, simplify=True, single_cls=False, source=None, split=val, stream_buffer=False, task=obb, time=None, tracker=botsort.yaml, translate=0.1, val=True, verbose=True, vid_stride=1, visualize=False, warmup_bias_lr=0.1, warmup_epochs=3.0, warmup_momentum=0.8, weight_decay=0.0005, workers=8, workspace=None
TensorBoard: Start with 'tensorboard --logdir /yolo/runs/obb/train4', view at http://localhost:6006/

                   from  n    params  module                                       arguments
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]
  2                  -1  1      6640  ultralytics.nn.modules.block.C3k2            [32, 64, 1, False, 0.25]
  3                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
  4                  -1  1     26080  ultralytics.nn.modules.block.C3k2            [64, 128, 1, False, 0.25]
  5                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
  6                  -1  1     87040  ultralytics.nn.modules.block.C3k2            [128, 128, 1, True]
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]
  8                  -1  1    346112  ultralytics.nn.modules.block.C3k2            [256, 256, 1, True]
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5, 3, True]
 10                  -1  1    249728  ultralytics.nn.modules.block.C2PSA           [256, 256, 1]
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 13                  -1  1    119808  ultralytics.nn.modules.block.C3k2            [384, 128, 1, True]
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 16                  -1  1     34304  ultralytics.nn.modules.block.C3k2            [256, 64, 1, True]
 17                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 19                  -1  1     95232  ultralytics.nn.modules.block.C3k2            [192, 128, 1, True]
 20                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 22                  -1  1    463104  ultralytics.nn.modules.block.C3k2            [384, 256, 1, True, 0.5, True]
 23        [16, 19, 22]  1    390360  ultralytics.nn.modules.head.OBB26            [15, 1, 1, True, [64, 128, 256]]
YOLO26n-obb summary: 291 layers, 2,652,984 parameters, 2,652,984 gradients, 6.3 GFLOPs

Transferred 792/792 items from pretrained weights
AMP: running Automatic Mixed Precision (AMP) checks...
AMP: checks passed ✅
train: Fast image access ✅ (ping: 0.0±0.0 ms, read: 2151.0±567.7 MB/s, size: 227.3 KB)
train: Scanning /yolo/datasets/dota8/labels/train... 4 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 4/4 712.4it/s 0.0s
train: New cache created: /yolo/datasets/dota8/labels/train.cache
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 1418.0±968.9 MB/s, size: 97.3 KB)
val: Scanning /yolo/datasets/dota8/labels/val... 4 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 4/4 1.2Kit/s 0.0s
val: New cache created: /yolo/datasets/dota8/labels/val.cache
Plotting labels to /yolo/runs/obb/train4/labels.jpg...
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically...
optimizer: MuSGD(lr=0.000526, momentum=0.9) with parameter groups 0 weight(decay=0.0), 0 weight(decay=0.0005), 24 bias(decay=0.0)
WARNING ⚠️ TensorBoard: TensorBoard graph visualization failure: Tracer cannot infer type of (tensor([[[ 5.1451e+01,  5.8704e+01,  8.8570e+01,  ...,  5.0365e-05,  7.0000e+00,  4.3944e-03],
         [ 7.6229e+01,  2.0657e+01,  1.1588e+02,  ...,  4.1951e-05,  7.0000e+00, -1.5607e-03],
         [ 5.1450e+01,  2.0277e+01,  8.3306e+01,  ...,  4.0920e-05,  7.0000e+00,  3.1936e-03],
         ...,
          [ 1.7078,  0.8693,  1.5679,  ..., -0.2693,  0.6455, -0.2170],
          [ 1.2936,  0.4024,  1.4049,  ..., -0.1055,  1.1518,  0.2598],
          [ 0.5727,  0.5314,  0.4425,  ...,  0.0149, -0.0866, -0.0448]]]], device='cuda:0')], 'angle': tensor([[[-0.0145, -0.0043,  0.0059,  ..., -0.0026,  0.0023, -0.0076]]], device='cuda:0', grad_fn=<CatBackward0>)}})
:Dictionary inputs to traced functions must have consistent type. Found Tensor and List[Tensor] -> Only Tensors created explicitly by the user (graph leaves) support the deepcopy protocol at the moment.  If you were attempting to deepcopy a module, this may be because of a torch.nn.utils.weight_norm usage, see https://github.com/pytorch/pytorch/pull/103001
Image sizes 320 train, 320 val
Using 8 dataloader workers
Logging results to /yolo/runs/obb/train4
Starting training for 10 epochs...
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss angle_loss  Instances       Size
       1/10      0.17G      1.524      3.222    0.04274    0.04499         52        320: 100% ━━━━━━━━━━━━ 4/4 2.1s/it 8.2s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 4/4 1.3it/s 3.0s
                   all          4          8      0.982      0.954      0.995      0.868
```
### Environment
```
Ultralytics 8.4.3 🚀 Python-3.11.14 torch-2.9.1+cu128 CUDA:0 (Tesla T4, 14913MiB)
Setup complete ✅ (80 CPUs, 62.3 GB RAM, 1084.2/3665.0 GB disk)

OS                     Linux-6.8.0-86-generic-x86_64-with-glibc2.35
Environment            Linux
Python                 3.11.14
Install                pip
Path                   /home/deeplearning/miniconda3/envs/yolo26/lib/python3.11/site-packages/ultralytics
RAM                    62.33 GB
Disk                   1084.2/3665.0 GB
CPU                    Intel Xeon Silver 4316 2.30GHz
CPU count              80
GPU                    Tesla T4, 14913MiB
GPU count              4
CUDA                   12.8

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.8>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 12.1.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.17.0>=1.4.1
torch                  ✅ 2.9.1>=1.8.0
torch                  ✅ 2.9.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.24.1>=0.9.0
psutil                 ✅ 7.2.1>=5.8.0
polars                 ✅ 1.37.1>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18
```
### Minimal Reproducible Example

```
yolo obb train model=yolo26n-obb.pt data=dota8.yaml epochs=10 imgsz=320 batch=1
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @kssion, thank you for your interest in Ultralytics 🚀! This is an automated response to help us triage quickly—an Ultralytics engineer will also assist soon 🙏

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.✅ Thanks for including an MRE already—if you can, please also paste the full command output from a fresh run and attach the relevant log directory contents from `/yolo/runs/obb/train4` (especially TensorBoard event files) so we can reproduce the TensorBoard graph visualization warning consistently.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by UltralyticsAssistant** [comment]
Thanks for reporting this — a potential fix has now been merged in the linked PR: [Fix TensorBoard error with OBB](https://github.com/ultralytics/ultralytics/pull/23276).

### What changed (and why it may fix this)
The PR updates the TensorBoard graph-logging callback to run under `smart_inference_mode()`:
- Imports `smart_inference_mode` from `ultralytics.utils.torch_utils`
- Wraps the TensorBoard graph export path (`_log_tensorboard_graph(trainer)`) with `@smart_inference_mode()` so it executes in an inference context (no gradient tracking)
- Scalar logging behavior is unchanged; only the model graph logging path is affected

This should make TensorBoard graph export safer and more efficient by avoiding autograd-related side effects, and may prevent the TensorBoard error seen with OBB while also reducing memory/overhead during graph logging.

### Credit
Merged PR authored by @Y-T-G with contributions from @lmycross — thanks for the fix and improvements.

### How to test the fix
There isn’t a specific version number noted in the PR description, so the easiest ways to validate are:

- Install from the latest `main`:
  - `pip install git+https://github.com/ultralytics/ultralytics.git@main`

- Or wait for the next official PyPI release and upgrade normally:
  - `pip install -U ultralytics`

### Feedback requested
If possible, please try one of the options above and share whether the TensorBoard + OBB issue is resolved on your setup (along with OS, Python, PyTorch, and `ultralytics` version). If it still reproduces, any logs/traceback details would help narrow down what’s left.

Thanks again 🙏 — and any further feedback is welcome if the issue persists.
